yt/utilities/on_demand_imports.py

```diff
@@ -251,6 +251,44 @@ def pooch(self):
 _pooch = pooch_imports()
 
 
+class pyart_imports:
+    _name = "pyart"
+
+    _pyart = None
+
+    @property
+    def pyart(self):
+        if self._pyart is None:
+            try:
+                import pyart as pyart
+            except ImportError:
+                pyart = NotAModule(self._name)
+            self._pyart = pyart
+        return self._pyart
+
+
+_pyart = pyart_imports()
+
+
+class xarray_imports:
+    _name = "xarray"
+
+    _xarray = None
+
+    @property
+    def xarray(self):
+        if self._xarray is None:
+            try:
+                import xarray as xr
+            except ImportError:
+                xarray = NotAModule(self._name)
+            self._xarray = xarray
+        return self._xarray
```

Comment:
sorry I'm late to review this.
Actually there is a slightly cleaner way define such classes that I advise you follow (it will make deduplicating that code a little easier if we ever do it).
Specifically I'm thinking of how it's done for  e.g. `f90nml_imports` `requests_imports`

Reply:
done